Adds a Test262-friendly implementation for use with --prelude

rwaldron opened the pull request:
Tests are not available in Test262, however when they become available, use the following:

(from within test262 directory)

```
test262-harness --prelude ../realms-polyfill/src/test262.js --hostType=jsc --hostPath=`which jsc` 'test/built-ins/ShadowRealm/**/*.js'
```

rwaldron committed:
Adds a Test262-friendly implementation for use with --prelude

Tests are not available in Test262, however when they become available, use the following:

(from within test262 directory)

test262-harness --prelude ../realms-polyfill/src/test262.js --hostType=jsc --hostPath=`which jsc` 'test/built-ins/Realm/**/*.js'

rwaldron committed:
chore: make sure src/test262.js isn't included in karma

rwaldron committed:
fix: don't throw on non-string inputs

rwaldron committed:
fix: throw on invalid realm object

rwaldron committed:
fix: importValue validates realm object

rwaldron committed:
fix: restore string validation

leobalter commented on src/test262.js:
```diff
@@ -0,0 +1,96 @@
+            }
+            throw new Error('importValue not supported');
+        }
+}
```
nit pick, missing whitespace

rwaldron replied:
So weird... github copilot keeps doing stupid things like changing the indentation of the next line :|

leobalter commented on src/test262.js:
```diff
@@ -0,0 +1,96 @@
+
+    Object.defineProperty(Realm.prototype, 'toString', {
+        value() {
+            return `[object Realm]`;
```
if you want more accuracy, you could add [`@@toStringTag`](https://tc39.es/proposal-realms/#sec-realm.prototype-@@tostringtag) here.

rwaldron committed:
fix: @@toStringTag

rwaldron committed:
rough implementation of import semantics

rwaldron committed:
module import caching

rwaldron committed:
refactor: make abstract operation names match spec

rwaldron committed:
refactor: more alignment with spec

rwaldron committed:
refactor: implement basic importValue for browser polyfill

rwaldron committed:
refactor: rename IsPrimitive -> IsPrimitiveOrCallable

rwaldron committed:
fix: remove "only" from tests

rwaldron committed:
fix: remove unused code from karma config

rwaldron committed:
refactor: implement realm making for d8, jsshell and jsc

rwaldron committed:
Realm -> ShadowRealm

rwaldron committed:
Spec alignment revisions

rwaldron committed:
Refactor: add a node.js specific polyfill and testing.

This may need further refactoring, but I can't figure out how to make the browser qunit tests work for node
